TypeSystem/flow-analysis/promotion_via_assignment_A05_t01.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via assignment of an
+/// expression of type `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and captured is false
+/// - and `S` is the current type of `x` in `VM`
+/// - and `T <: S` and not `S <: T`
+/// - and `T` is a type of interest for `x` in `tested`
+///
+/// @description Checks that if a type of variable is not specified and there is
+/// no initializer then its type is `dynamic` in spite of assignment.
+/// @author sgrekhov22@gmail.com
+
+import "../../Utils/expect.dart";
+
+class S {}
+
+class T extends S {
+  int foo() => 42;
+}
+
+main() {
+  var x;
+  x = S();
+  if (x is T) {} // Make `T` a type of interest
+  x = T();
+  x.foo();
+  Expect.throws(
+    () {x.bar();},  // Check that `x` is dynamic
```

Comment:
I think `x` has been promoted to type `T` in the body of `main`. However, this occurrence of `x` is in a function literal, which means that it cannot be promoted (because flow analysis doesn't make any attempts to understand exactly at which locations in code this function literal could be executed). So `x` isn't promoted to type `T` in the body of the function literal, it's still of type `dynamic`.

This means that the test in the function literal says nothing which can be connected to "in spite of assignment".

Perhaps this is an updated version of an old test, written way back when a `dynamic` variable was actually never promoted?

Reply:
Yes, exactly! This is an updated version of the old test. Fixed. PTAL.

Reply:
Thanks!